reference-architectures/cloud_deploy_flow/deploy.tf

```diff
@@ -0,0 +1,59 @@
+# Assign Cloud Deploy Admin role to the default service account
+resource "google_project_iam_member" "cloud_deploy_admin_binding" {
+  project = var.project_id
+  role    = "roles/clouddeploy.admin"
+  member  = "serviceAccount:${data.google_compute_default_service_account.default.email}"
+}
+
+# Assign Cloud Deploy Releaser role to the default service account
+resource "google_project_iam_member" "cloud_deploy_releaser" {
+  project = var.project_id
+  role    = "roles/clouddeploy.releaser"
+  member  = "serviceAccount:${data.google_compute_default_service_account.default.email}"
+}
+
+# Grant "Service Account User" role to the default Compute Engine service account on the Cloud Build service account
+# Required for Cloud Functions to handle releases (Maybe? Probably isn't needed)
+resource "google_service_account_iam_binding" "allow_compute_sa_to_act_as" {
+  service_account_id = google_service_account.cloudbuild_service_account.name
+  role               = "roles/iam.serviceAccountUser"
+
+  members = [
+    "serviceAccount:${data.google_compute_default_service_account.default.email}",
+  ]
+}
+
+# Create a Cloud Deploy pipeline
+resource "google_clouddeploy_delivery_pipeline" "primary" {
+  name        = "random-date-service"
+  project = var.project_id
```

Comment:
Consider using the [`google_project` datasource](https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/project.html) instead of using `var.project_id` directly so you can check if a project exists before using its ID as a reference.

If you go this route, check the other Terraform resources as well for this.